src/core/thread/mle_router.cpp

```diff
@@ -1426,127 +1426,114 @@ void MleRouter::UpdateRoutes(const RouteTlv &aRoute, uint8_t aRouterId)
     uint8_t oldNextHop;
     uint8_t cost;
     uint8_t linkQuality;
-    bool    update;
     bool    resetAdvInterval = false;
     bool    changed          = false;
+    uint8_t myRouterId;
 
     neighbor = mRouterTable.GetRouter(aRouterId);
     VerifyOrExit(neighbor != NULL);
 
-    // update routes
-    do
+    // update link quality out to neighbor
+    myRouterId = GetRouterId(GetRloc16());
+    if (myRouterId <= kMaxRouterId && aRoute.IsRouterIdSet(myRouterId))
     {
-        update = false;
+        uint8_t myRouteCount = 0;
+        for (uint8_t i = 0; i < myRouterId; i++)
+        {
+            if (aRoute.IsRouterIdSet(i))
```

Comment:
why not `myRouteCount+=aRoute.IsRouterIdSet(i)`?

Reply:
sure, why not. changed.